### PR TITLE
[283] Eval Form: Cancelation modal

### LINKED ISSUE #283: Eval Form: Cancelation modal
# What/Why
When a user cancels the Evaluation Form form entry, they should see a modal popup that warns them their progress will be lost.

### Figma Link
Image

## Acceptance Criteria
- [x] Modal displays when the user cancels from a newly created form
- [x] Modal displays when the user cancels from an edited form

_Disregard the following if this work **does not** include any UI changes_
- [x] WAVE screenshot
- [x] Automated a11y tests (axe-rspec specs)
- [x] Keyboard accessibility (tab navigation, form interactions)
- [x] Responsiveness at desktop and mobile

## Out of Scope
Not checking "dirty" state of the form

### PR DESCRIPTION
**Pull request by cpreisinger**
#283 

Adds a new modal for confirming redirects/actions

### CODE REVIEW

File: app/javascript/confirmation_modal.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+document.addEventListener("DOMContentLoaded", () => {
```
Comment (stepchud) [inline app/javascript/confirmation_modal.js]: can we convert this to a stimulus controller instead? we have a few modal controllers there already.

File: app/javascript/confirmation_modal.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const modalTriggers = document.querySelectorAll("[data-modal]");
+
+  modalTriggers.forEach((trigger) => {
+    trigger.addEventListener("click", (e) => {
+      console.log(e);
```
Comment (stepchud) [inline app/javascript/confirmation_modal.js]: console.log

**Review by stepchud — CHANGES_REQUESTED** [review]
I like the reusability of this for any other modals that just do confirmation with a redirect, but I think for consistency it should be stimulus. Let me know if that doesn't make sense here or you disagree.

File: spec/system/evaluation_form_spec.rb [hunk 3]
```diff
@@ -16,6 +16,45 @@
       expect(page).to(be_axe_clean)
     end
 
+    it "shows a confirmation modal when clicking the cancel button" do
```
Comment (stepchud) [inline spec/system/evaluation_form_spec.rb:19]: nice specs! 🤘

**Comment by cpreisinger** [comment]
Codeclimate failure for "similar" code blocks in the javascript. Should we suppress this warning?

File: app/views/modals/_confirmation.erb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<dialog
+  id="<%= id %>"
+  class="border-width-0 radius-lg"
+  aria-labelledby=<%= "#{id}-heading" %>
+  aria-describedby=<%= "#{id}-description" %>
+  data-modal-target="modal"
+  data-modal-confirm-redirect="<%= defined?(confirm_redirect) ? confirm_redirect : '' %>"
+  data-modal-cancel-redirect="<%= defined?(cancel_redirect) ? cancel_redirect : '' %>"
+  data-modal-confirm-action="<%= defined?(confirm_action) ? confirm_action : '' %>"
+  data-modal-cancel-action="<%= defined?(cancel_action) ? cancel_action : '' %>"
```
Comment (stepchud) [inline app/views/modals/_confirmation.erb:10]: nice!

**Review by stepchud — APPROVED** [review]
very 🆒 😎

**Pull request merged**